### PR TITLE
fix: empty token should be invalid

### PR DESCRIPTION
**Pull request by pauldambra**
see https://posthoghelp.zendesk.com/agent/tickets/9991 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/12757)

a user was unintentionally initializing posthog with the empty string as a token, and the empty string as an api_host

we silently did nothing

let's warn the user and bail on setup when the token is the empty string

### CODE REVIEW

**Review by benjackwhite — APPROVED** [review]
I feel like I did something like this before and got push back on it but I can't recall properly what it was 😅  so Maybe its fine?

File: src/posthog-core.ts [hunk 1]
```diff
@@ -379,6 +380,12 @@ export class PostHog {
             logger.critical('You must name your new library: init(token, config, name)')
             return
         }
+
+        if (_isUndefined(token) || _isEmptyString(token)) {
+            logger.critical('You must provide a token to initialize posthog')
+            return
+        }
```
Comment (marandaneto) [inline src/posthog-core.ts:387]: > a user was unintentionally initializing posthog with the empty string as a token, and the empty string as an api_host

should we verify the `api_host` as well? the check is only for `token`.

Comment (marandaneto) [reply src/posthog-core.ts:387]: The `api_host` we can also check if it's a valid `URL` as well.

Comment (pauldambra) [reply src/posthog-core.ts:387]: I can almost imagine a situation where the empty string is valid (A single page, a single URL, posthog hosted on the same url so `/decide` is always valid) - very edge case but 🙈 

Will ignore URL validation here to keep the PR smaller :)

**Pull request merged**